roles/docker/tasks/main.yml

```diff
@@ -1,32 +1,43 @@
 ---
 - name: Ensure old versions of Docker are not installed
-  package:
+  ansible.builtin.package:
     name:
       - docker
       - docker.io
       - docker-engine
     state: absent
+  tags: docker
 
-- name: Add Docker apt key
-  get_url:
+- name: Create directory for Docker GPG key
+  ansible.builtin.file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: "0755"
+  tags: docker
+
+- name: Add Docker GPG apt key
+  ansible.builtin.get_url:
     url: "https://download.docker.com/linux/ubuntu/gpg"
-    dest: /etc/apt/trusted.gpg.d/docker.asc
-    mode: 0644
-    force: false
+    dest: /etc/apt/keyrings/docker.asc
```

Comment:
This isn't arbitrary. Got some errors locally on 24.04.1 and the [official docs](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository) store the gpg key in `/etc/apt/keyrings`:

```bash
# Add Docker's official GPG key:
sudo apt-get update
sudo apt-get install ca-certificates curl
sudo install -m 0755 -d /etc/apt/keyrings
sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
sudo chmod a+r /etc/apt/keyrings/docker.asc
```